work_queue/src/work_queue.c

```diff
@@ -4569,29 +4570,39 @@ static int send_one_task( struct work_queue *q )
 	struct work_queue_task *t;
 	struct work_queue_worker *w;
 
+	int tasks_considered = 0;
 	timestamp_t now = timestamp_get();
 
-	// Consider each task in the order of priority:
-	list_first_item(q->ready_list);
-	while( (t = list_next_item(q->ready_list))) {
```

Comment:
I'm a little uncomfortable with the implied free/malloc that comes from pop/push every time we consider one item.  Also, the multiple exit points that all must call `list_push_tail` looks like a maintenance hazard.

What would you think of introducing a new call `list_rotate` that moved the head item to the tail (without free/malloc) returning its value?

So then you would have this:
```
while( t= list_rotate(q->ready_list) ) {
    // check various conditions and return if needed

   // finally
   list_pop_tail
   commit_task_to_worker
}
```